pkg/clients/edgex-foundry/util.go

```diff
@@ -44,20 +44,25 @@ type ClientURL struct {
 	Port int
 }
 
-func getEdgeDeviceName(d *devicev1alpha1.Device) string {
+type EdgeNameProvider interface {
+	GetObjectMeta() metav1.Object
+	GetName() string
+}
+
+func getEdgeName(provider EdgeNameProvider) string {
```

Comment:
Perhaps getEdgeXName is better since this util is dedicated for EdgeX Foundry?

Reply:
good suggestion